lib/thousand_island/server.ex

```diff
@@ -5,7 +5,7 @@ defmodule ThousandIsland.Server do
 
   @spec start_link(ThousandIsland.ServerConfig.t()) :: Supervisor.on_start()
   def start_link(%ThousandIsland.ServerConfig{} = config) do
-    Supervisor.start_link(__MODULE__, config)
+    Supervisor.start_link(__MODULE__, config, config.supervisor_options)
```

Comment:
Not really important but if we don't need these options passed through `init`/etc, maybe this?
```suggestion
    {supervisor_options, config} = Keyword.pop(config, :supervisor_options, [])
    Supervisor.start_link(__MODULE__, config, supervisor_options)
```

Reply:
Sure!

Reply:
Enh, we don't do that with any other options, and since config is a full blown `ServerConfig` struct I'm inclined to pass it through unchanged for consistency

Reply:
makes sense